algebra-testing/src/main/java/com/hubspot/assertj/algebra/error/ResultShouldBeErrWithValue.java

```diff
@@ -10,11 +10,13 @@ private ResultShouldBeErrWithValue(String format, Object... arguments) {
     super(format, arguments);
   }
 
-  public static <T, E> ErrorMessageFactory shouldBeOkWithValue(Result<T, E> actual, Object value) {
+  public static <T, E> ErrorMessageFactory shouldBeErrWithValue(Result<T, E> actual, Object error) {
     if (actual.isErr()) {
-      return new ResultShouldBeErrWithValue("Expecting Result to be Err containing <%s> but contained <%s>", value, actual.unwrapErrOrElseThrow());
+      E actualError = actual.unwrapErrOrElseThrow();
+      return new ResultShouldBeErrWithValue("Expecting Result to be Err containing <%s> but contained <%s>. expected:<%s> but was:<%s>", error, actualError, error, actualError);
     } else {
-      return new ResultShouldBeErrWithValue("Expecting Result to be Err containing <%s> but was Ok containing <%s>", value, actual.unwrapOrElseThrow());
+      return new ResultShouldBeErrWithValue("Expecting Result to be Err containing <%s> but was Ok containing <%s>", error, actual.unwrapOrElseThrow());
```

Comment:
No need to do the same for this case imo. If the actual Result is `ok` while we're expecting an `err `result, the two objects to compare are two different things conceptually.